spec.md

```diff
@@ -582,6 +604,12 @@ message ControllerPublishVolumeRequest {
   // Whether to publish the volume in readonly mode. This field is
   // REQUIRED.
   bool readonly = 5;
+
+  // OPTIONAL field for storing principal information to be used in
+  // the publishing of this volume. If the principal is present the
+  // plugin MAY ensure the volume published will be accessible for
+  // the herein specified entities only.
+  Principal container_acts_as = 6;
```

Comment:
The `container_acts_as` values must be identical.
Personally speaking our plugin will use only the `NodePublishVolume` principal field but others may require preparatory steps and need this information in the `ControllerPublishVolume` call, too.

I'll rename the field to `accessed_as` ,  as the principal describes the acting id for a given binding. One volume can be published in different containers with different principals at the same time, resulting in different containers acting as different principals on the files in that volume.
This also relates to issue #40 .